artemis/modules/nuclei.py

```diff
@@ -412,10 +422,43 @@ def run_multiple(self, tasks: List[Task]) -> None:
 
         self.log.info(f"running {len(templates)} templates and {len(self._workflows)} workflow on {len(tasks)} hosts.")
 
-        targets = []
+        targets: List[str] = []
         for task in tasks:
             targets.append(get_target_url(task))
 
+        tech_results = run_tech_detection(targets, self.log)
+
```

Comment:
lines 428-448 are quite complex, what do you think about extracting them to a function and adding a test for them?

in docker-compose.test.yml we already have multiple services - wordpress, joomla, ... to choose from for technlogy detection